### ISSUE TITLE
interpolation calls the old interpolation implementaion which produces warnings, could be considered error due to could predictable execution path. 

### ISSUE THREAD

**Issue by karimnorouzi** [firedrakeproject/firedrake#3383]
minimum code to reproduce the warning messages. 

```py
from firedrake import UnitSquareMesh, FunctionSpace, SpatialCoordinate, assemble
from firedrake.__future__ import interpolate

m1 = UnitSquareMesh(16,16)
m2 = UnitSquareMesh(24, 24);
W = FunctionSpace(m1, 'CG', 1)
Q = FunctionSpace(m2, 'CG', 1)
x, z = SpatialCoordinate(m1)
T = assemble(interpolate(x + z, W))
Tnu = assemble(interpolate(T, Q))
```

**Comment by connorjward** [comment]
@nbouziani you would definitely be best suited to fixing this since you introduced `firedrake.__future__`.

**Comment by nbouziani** [comment]
Fixed in #3390.
